daras_ai_v2/base.py

```diff
@@ -2425,6 +2450,42 @@ def is_current_user_paying(self) -> bool:
     def is_current_user_owner(self) -> bool:
         return bool(self.request.user and self.current_sr_user == self.request.user)
 
+    def is_user_authorized(self, user: AppUser | None = None) -> bool:
+        if self.is_user_admin(user):
+            return True
+
+        sr, pr = self.current_sr_pr
+        if pr.saved_run_id != sr.id:
```

Comment:
i think this is a bit too restrictive and we should only do this for paid workspaces. i.e. personal and non paid workspaces' runs are public by default